src/hardware/pcspeaker.cpp

```diff
@@ -21,16 +21,41 @@
 #include <algorithm>
 #include <array>
 #include <cmath>
+#include <limits>
 
 #include "timer.h"
 #include "setup.h"
 #include "pic.h"
 
 #define SPKR_ENTRIES 1024
-#define SPKR_POSITIVE_VOLTAGE 5000.0f
-#define SPKR_NEUTRAL_VOLTAGE  0.0f
-#define SPKR_NEGATIVE_VOLTAGE -SPKR_POSITIVE_VOLTAGE
-#define SPKR_SPEED (SPKR_POSITIVE_VOLTAGE * 2.0f / 0.070f)
+#define SPKR_SPEED   (AMPLITUDE_POSITIVE * 2.0f / 0.070f)
+
+/*
+Amplitude levels for the speaker are computed as follows:
+1. Start with the maximum allow level
+2. Reduce it by -5 LUFS to match generally upper-bound normalization
+   levels for normal audio. Knowing that the PC speaker hits peak
+   amplitude, we simply take off 5 dB.
+2. Given the PC speaker's PWM-mode doesn't actually produce sine
+   curves and instead are stepped approximations, we drop this by the
+   RMS level of a sine curve, which should provide roughly the same
+   perceived loudness.
+3. When the PC Speaker generates square waves, we apply the above reductions
+   again plus an additional RMS factor, knowing that square wave are
+   exclusively at full amplitude, and carry twice the power of sine waves.
+
+TODO: sqrt() and log() are not available as a constexpr in C++14,
+      so hard-code them for now.
+*/
+constexpr float AMPLITUDE_MINUS_5DB = 0.562341f;
+constexpr float AMPLITUDE_RMS = static_cast<float>(M_SQRT1_2);
+constexpr float AMPLITUDE_POSITIVE = std::numeric_limits<int16_t>::max() *
+                                     AMPLITUDE_MINUS_5DB * AMPLITUDE_RMS;
+constexpr float AMPLITUDE_NEGATIVE = std::numeric_limits<int16_t>::min() *
+                                     AMPLITUDE_MINUS_5DB * AMPLITUDE_RMS;
+constexpr float AMPLITUDE_NEUTRAL = (AMPLITUDE_POSITIVE + AMPLITUDE_NEGATIVE) / 2;
+constexpr float AMPLITUDE_SQUARE_WAVE_REDUCER = AMPLITUDE_MINUS_5DB *
+                                                AMPLITUDE_RMS * AMPLITUDE_RMS;
```

Comment:
Isn't this simply `AMPLITUDE_MINUS_5DB / 2`?

Reply:
Yup (RMS^2 == 0.5), but I wanted to show that we're reducing the volume by RMS 'twice over".
I've added this explanation to the comment block, and now simply `/ 2` in the expression.